beluga_tutorial/src/main.cpp

```diff
@@ -0,0 +1,258 @@
+// Copyright 2022-2024 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// https://github.com/Ekumen-OS/beluga/issues/279#issuecomment-1903914387
+
+#include <cmath>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <string>
+#include <vector>
+
+#include <range/v3/action.hpp>
+#include <range/v3/algorithm/any_of.hpp>
+#include <range/v3/algorithm/min_element.hpp>
+#include <range/v3/numeric/accumulate.hpp>
+#include <range/v3/range/conversion.hpp>
+#include <range/v3/view/generate.hpp>
+#include <range/v3/view/take_exactly.hpp>
+
+#include <yaml-cpp/yaml.h>
+#include <beluga/beluga.hpp>
+
+// TODO(alon): Explain each parameter.
```

Comment:
@Alondruck +1

Reply:
Should I describe each parameter using the following style?
/// Name of parameter
/**
* Description
*/

Reply:
You can check [Doxygen documentation](https://www.doxygen.nl/manual/docblocks.html) on _putting documentation after members_ (you can Ctrl + F that).